### PR TITLE
[DO NOT REVIEW] [skip ci] Unbacked SymInt mega-PR

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #94735

[skip ci]

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

Differential Revision: [D43237216](https://our.internmc.facebook.com/intern/diff/D43237216)

### CODE REVIEW

**Comment by ezyang** [comment]
@ezyang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D43237216).

File: c10/core/TensorImpl.cpp [hunk 1]
```diff
@@ -260,8 +309,18 @@ SymBool TensorImpl::compute_contiguous(identity<SymBool>) const {
...
-  return _compute_contiguous<c10::SymInt>(
-      extra_meta_->sizes_, extra_meta_->strides_, extra_meta_->numel_);
+  SymIntArrayRef sizes = extra_meta_->sizes_;
+  SymIntArrayRef strides = extra_meta_->strides_;
+  auto n = normalize_sym_sizes_strides(sizes, strides);
+  if (n.has_value()) {
+    SymNode base;
+    std::vector<SymNode> size_nodes;
+    std::vector<SymNode> stride_nodes;
+    std::tie(base, size_nodes, stride_nodes) = *n;
```
Comment (Skylion007) [inline c10/core/TensorImpl.cpp:319]: I know you said not to review this, but here is no need to use std::tie now that we are on C++17 with structure bindings ;)

Comment (ezyang) [reply c10/core/TensorImpl.cpp:319]: Can't use it, our Windows envs choke on it :(

Comment (Skylion007) [reply c10/core/TensorImpl.cpp:319]: Darn...

**Pull request closed without merging**